### PR TITLE
Add cls embeddings

### PR DESCRIPTION
**Pull request by AdamIzdebski**

### CODE REVIEW

**Comment by AdamIzdebski** [comment]
@jano1906 note that cls_embeddings and lm_embeddings have a non-empty intersection in this version of Jointformer

File: jointformer/models/utils.py [hunk 1]
```diff
@@ -55,47 +17,30 @@ def to(self, device: str, pin_memory: bool = True) -> 'ModelInput':
...
+        if _is_cls_token_embedding:
+            return self['cls_embeddings']
+        elif self['attention_mask'] is not None:
+            w = self['attention_mask'] / self['attention_mask'].sum(dim=-1, keepdim=True)
             w = w.unsqueeze(-2)
-            global_embedding = w @ self.embeddings
+            global_embedding = w @ self['cls_embeddings']
             return global_embedding.squeeze(-2)
+        else:
+            return self['cls_embeddings'].mean(dim=-1)
```
Comment (jano1906) [inline jointformer/models/utils.py:37]: Why _is_cls_token_embedding requires len(cls_ebedding) == 2? Seems odd. Also note that len(tensor) returns tensor.shape[0] so it's the batch size. I don't know what's happening here.

if we don't have cls_embeddings we want to aggregate the lm_embeddings with respect to attention mask, don't change the meaning of cls_embedding based on the count of their dimension. The entity constructing ModelOutput should decide which embeddings are cls and which are lm and if it does not produce cls_embeddings, then ModelOutput should automatically aggregate lm_embeddings based on attention mask.

**Review by jano1906 — CHANGES_REQUESTED** [review]
https://github.com/szczurek-lab/jointformer/pull/35/files#r1709721948

File: jointformer/models/utils.py (hunk 1 above)
Comment (AdamIzdebski) [reply jointformer/models/utils.py]: changed

Comment (AdamIzdebski) [reply jointformer/models/utils.py]: @jano1906 do you want to change to dataclasses in this or in a new pr?

**Review by jano1906 — APPROVED** [review]
